compose/project.py

```diff
@@ -390,13 +390,18 @@ def do(service):
         def get_deps(service):
             return {self.get_service(dep) for dep in service.get_dependency_names()}
 
-        results = parallel.parallel_execute(
+        results, errors = parallel.parallel_execute(
             services,
             do,
             operator.attrgetter('name'),
             None,
             get_deps
         )
+        if errors:
+            raise ProjectError(
+                'Encountered errors while bringing up the project.'
+            )
```

Comment:
The detailed error with service information is reported during the `parallel_execute` call ( https://github.com/docker/compose/blob/master/compose/parallel.py#L57 )